internal/core/entity/filter.go

```diff
@@ -0,0 +1,53 @@
+package entity
+
+func getUpstreamDef(obj interface{}) (uf *UpstreamDef) {
+	switch obj := obj.(type) {
+	case *Upstream:
+		uf = &obj.UpstreamDef
+	case *Service:
+		uf = obj.Upstream
+	case *Route:
+		uf = obj.Upstream
+	default:
+		return nil
+	}
+	return
+}
+
+func ServiceFilter(obj interface{}) bool {
```

Comment:
Those ServiceXXX functions are defined but unused?

Reply:
These functions will be used in the Watcher component.

Reply:
Would you split them out and submit them in another PR? Thanks!